tests/test_classifier.py

```diff
@@ -102,6 +101,27 @@ def test_fit_lm_only(self):
         for proba in probabilities:
             self.assertIsInstance(proba, dict)
 
+    def test_cached_predict(self):
+        """
+        Ensure second call to predict is faster than first
+        """
+
+        model = Classifier(config=self.default_config())
+        train_sample = self.dataset.sample(n=self.n_sample)
+        valid_sample = self.dataset.sample(n=self.n_sample)
+        model.fit(train_sample.Text.values, train_sample.Target.values)
+
+        start = time.time()
+        model.predict(valid_sample.Text[:1].values)
+        first = time.time()
+        model.predict(valid_sample.Text[:1].values)
+        second = time.time()
+
+        first_prediction_time = (first - start)
+        second_prediction_time = (second - start)
+        self.assertLess(second_prediction_time, first_prediction_time / 2.)
+
+
     def test_fit_predict(self):
```

Comment:
Need a test that goes fit predict, fit predict and ensure that the 2nd fit call does not cause issues, and that the second predictions are different than the first ones. I suspect that this will fail.